### PR TITLE
Allow passing a single "url" option

### PR DESCRIPTION
**Pull request by rfk**
This contains some refactoring I did in support of https://github.com/mozilla/fxa-auth-server/pull/1070.  It allows the oauth server to be specified as a single url rather than independent (host, port) values.  I also did a bit of general cleanup e.g. removing global state, adding licence boilerplate, etc.

No tests here, but i tested it with https://github.com/mozilla/fxa-auth-server/pull/1070 using both `oauth.host` and `oauth.url` configuration options and it all seemed to work OK.

/cc @seanmonstar; @dannycoates r?

### CODE REVIEW

**Comment by rfk** [comment]
Also if this seems like an OK change, can we please get a new release with it in?

File: index.js [hunk 1]
```diff
@@ -1,18 +1,16 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+var url = require('url')
 var boom = require('boom')
 var Pool = require('poolee')
-var pool = null
```
Comment (seanmonstar) [inline index.js:3]: \o/

**Comment by rfk** [comment]
> can we please get a new release with it in?

Oh sweet, actually I have the power to do this myself

**Comment by dannycoates** [comment]
good stuff @rfk!

**Pull request merged**